dex/testing/eth/create-node.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/sh
+# Script for creating eth nodes.
+set -e
+
+# The following are required script arguments
+TMUX_WIN_ID=$1
+NAME=$2
+NODE_PORT=$3
+CHAIN_ADDRESS=$4
+CHAIN_PASSWORD=$5
+CHAIN_ADDRESS_JSON=$6
+CHAIN_ADDRESS_JSON_FILE_NAME=$7
+ADDRESS=$8
+ADDRESS_PASSWORD=$9
+ADDRESS_JSON=${10}
+ADDRESS_JSON_FILE_NAME=${11}
+NODE_KEY=${12}
+
+GROUP_DIR="${NODES_ROOT}/${NAME}"
+MINE_JS="${GROUP_DIR}/mine.js"
+NODE_DIR="${GROUP_DIR}/node"
+mkdir -p "${NODE_DIR}"
+
+# Write node ctl script.
+cat > "${NODES_ROOT}/harness-ctl/${NAME}" <<EOF
+#!/bin/sh
+geth --datadir="${NODE_DIR}" \$*
+EOF
+chmod +x "${NODES_ROOT}/harness-ctl/${NAME}"
+
+# Write mine script.
+cat > "${NODES_ROOT}/harness-ctl/mine-${NAME}" <<EOF
+#!/bin/sh
+  case \$1 in
+      ''|*[!0-9]*)  ;;
+      *) NUM=\$1 ;;
+  esac
+  for i in \$(seq \$NUM) ; do
+    ./${NAME} attach --preload "${MINE_JS}" --exec 'mine()'
+  done
+EOF
+chmod +x "${NODES_ROOT}/harness-ctl/mine-${NAME}"
+
+# Write mining javascript.
+# NOTE: This sometimes mines more than one block. It is a race. This returns
+# the number of blocks mined within the lifespan of the function, but one more
+# MAY be mined after returning.
+cat > "${MINE_JS}" <<EOF
+function mine() {
+  blkN = eth.blockNumber;
+  miner.start();
+  miner.stop();
+  admin.sleep(1.1);
+  return eth.blockNumber - blkN;
+}
+EOF
+
+# Write password file to unlock accounts later.
+cat > "${GROUP_DIR}/password" <<EOF
+$CHAIN_PASSWORD
+$ADDRESS_PASSWORD
+EOF
+
+cat > "${NODE_DIR}/eth.conf" <<EOF
+[Eth]
+NetworkId = 42
+SyncMode = "full"
+
+[Eth.Miner]
+Etherbase = "0x${CHAIN_ADDRESS}"
+
+[Eth.Ethash]
+DatasetDir = "${NODE_DIR}/.ethash"
+
+[Node]
+DataDir = "${NODE_DIR}"
+
+[Node.P2P]
+NoDiscovery = true
+BootstrapNodes = []
+BootstrapNodesV5 = []
+ListenAddr = ":${NODE_PORT}"
+NetRestrict = [ "127.0.0.1/32" ]
```

Comment:
Can we add `::1` to this list? And maybe change the ipv4 loopback network to 127.0.0.0/8

Reply:
Ok.